builtin/providers/docker/resource_docker_container.go

```diff
@@ -341,6 +345,54 @@ func resourceDockerContainer() *schema.Resource {
 				Elem:     &schema.Schema{Type: schema.TypeString},
 				Set:      stringSetHash,
 			},
+
+			"uploads": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"local_path": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: func(v interface{}, k string) (ws []string, es []error) {
+								path := v.(string)
+								if _, err := os.Stat(path); err != nil {
+									es = append(es, fmt.Errorf(
+										"%q must be valid path: %s", path, err))
+								}
+								return
+							},
+							StateFunc: func(v interface{}) string {
```

Comment:
I'm not sure why but during my tests, it turns out that the generated state file leaves the path here but while doing a plan, the checksum is generated. This leads to changes such as

```
    uploads.#:                         "1" => "1"
    uploads.241032095.local_path:      "files/" => ""
    uploads.241032095.remote_path:     "/tmp/" => ""
    uploads.4134509434.local_path:     "" => "7c95cb15f351188c8a680e2a2e15df87d163cc8a" (forces new resource)
    uploads.4134509434.remote_path:    "" => "/tmp/" (forces new resource)
```

Could this be related to the many issues related to `StateFunc` that cropped up everywhere :(